lib/extended-page-mod.js

```diff
@@ -333,92 +341,90 @@ const ExtendedPageMod = Loader.compose(EventEmitter, {
     );
   },
 
-  _unregisterStyleSheet : function () {
+  _unregisterStyleSheet() {
     let uri = this._registeredStyleURI;
 
-    if (uri  && styleSheetService.sheetRegistered(uri, USER_SHEET))
+    if (uri && styleSheetService.sheetRegistered(uri, USER_SHEET)) {
       styleSheetService.unregisterSheet(uri, USER_SHEET);
+    }
 
     this._registeredStyleURI = null;
   }
 });
-exports.ExtendedPageMod = function(options) ExtendedPageMod(options)
+exports.ExtendedPageMod = options => ExtendedPageMod(options);
 exports.ExtendedPageMod.prototype = ExtendedPageMod.prototype;
 
-const Registry = EventEmitter.compose({
-  _registry: null,
-  _constructor: null,
-  constructor: function Registry(constructor) {
-    this._registry = [];
-    this._constructor = constructor;
-    this.on('error', this._onError = this._onError.bind(this));
-    unload.ensure(this, "_destructor");
-  },
-  _destructor: function _destructor() {
-    let _registry = this._registry.slice(0);
-    for (let instance of _registry)
-      this._emit('remove', instance);
-    this._registry.splice(0);
-  },
-  _onError: function _onError(e) {
-    if (!this._listeners('error').length)
-      console.error(e);
+/**
+ * ExtendedPageModManager manages all registed instances of ExtendedPageMod
+ * It does most of this work on `document-element-inserted`
+ * and then finding the instance associated with a window 
+ */
+const ExtendedPageModManager = Class({
+  extends: EventTarget,
+  implements: [Disposable],
+  _off: EventTarget.prototype.off,
+  _onError(e) {
+    if (!count(this, 'error')) console.error(e);
   },
-  has: function has(instance) {
+  has(instance) {
     let _registry = this._registry;
     return (
-      (0 <= _registry.indexOf(instance)) ||
-      (instance && instance._public && 0 <= _registry.indexOf(instance._public))
+      (0 <= _registry.indexOf(instance)) /*||
+        (instance && instance._public && 0 <= _registry.indexOf(instance._public))//*/
```

Comment:
So in the old `compose`/`resolve`, it appeared to be the real instance of the class. You couldn't do `fluffy instanceof Dog`. You'd have to do `fluffy._public instanceofDog`.